exonum-java-binding/core/rust/src/runtime/cmd.rs

```diff
@@ -15,150 +15,166 @@
+            Command::Run(c) => c.execute(),
+            Command::Maintenance(c) => c.execute().map(Into::into),
+        }
     }
+}
+
+/// EJB-specific `run` command which collects standard Exonum Core parameters and
+/// also additional Java runtime and JVM configuration parameters.
+#[derive(Debug, StructOpt, Serialize, Deserialize)]
+#[structopt(rename_all = "kebab-case")]
+pub struct Run {
+    #[structopt(flatten)]
+    #[serde(flatten)]
+    standard: StandardRun,
+    /// A port of the HTTP server for Java services.
+    ///
+    /// Must be distinct from the ports used by Exonum.
+    #[structopt(long)]
+    ejb_port: i32,
+    /// Path to the directory containing service artifacts.
+    #[structopt(long)]
+    artifacts_path: PathBuf,
+    /// Path to log4j configuration file.
+    #[structopt(long)]
+    ejb_log_config_path: Option<PathBuf>,
+    /// Overrides the standard path to native libraries, enabling running the non-packaged
+    /// exonum-java application.
+    ///
+    /// Mostly for internal usage.
+    #[structopt(long)]
+    ejb_override_java_library_path: Option<PathBuf>,
+    /// Allows JVM being remotely debugged.
+    ///
+    /// Takes a socket address as a parameter in form of `HOSTNAME:PORT`.
+    /// For example, `localhost:8000`
+    #[structopt(long)]
+    jvm_debug: Option<String>,
+    /// Additional parameters for JVM that precede the rest of arguments.
+    ///
+    /// Must not have a leading dash. For example, `Xmx2G`.
+    #[structopt(long)]
+    jvm_args_prepend: Vec<String>,
+    /// Additional parameters for JVM that get appended to the rest of arguments.
+    ///
+    /// Must not have a leading dash. For example, `Xmx2G`.
+    #[structopt(long)]
+    jvm_args_append: Vec<String>,
+}
 
-    fn execute(&self, mut context: Context) -> Result<Context, failure::Error> {
-        let log_config_path = context
-            .arg(EJB_LOG_CONFIG_PATH)
-            .unwrap_or_else(|_| get_path_to_default_log_config());
-        let port = context.arg(EJB_PORT)?;
-        let override_system_lib_path = context.arg(EJB_OVERRIDE_LIBRARY_PATH).ok();
-        let args_prepend: Vec<String> = context.arg_multiple(JVM_ARGS_PREPEND).unwrap_or_default();
-        let args_append: Vec<String> = context.arg_multiple(JVM_ARGS_APPEND).unwrap_or_default();
-        let jvm_debug_socket = context.arg(JVM_DEBUG_SOCKET).ok();
-
-        let jvm_config = JvmConfig {
-            args_prepend,
-            args_append,
-            jvm_debug_socket,
-        };
-
-        let runtime_config = RuntimeConfig {
-            log_config_path,
-            port,
-            override_system_lib_path,
-        };
-
-        let config = Config {
-            jvm_config,
-            runtime_config,
-        };
-
-        // Write the final EJB configuration.
-        write_ejb_config(&mut context, Value::try_from(config)?);
-        Ok(context)
+/// Possible output of the Java Bindings CLI commands.
```

Comment:
Can we now specify the application name and version in the help output? (ECR-3090)

Reply:
Yep! It is already done, see the output of running exonum-java without arguments:
```
     Running `/Users/ilyabogdanov/projects/exonum-java-binding/exonum-java-binding/core/rust/target/debug/exonum-java`
java-bindings 0.9.0-SNAPSHOT
Exonum Team <exonum@bitfury.com>
Exonum Java Bindings Application.

Configures and runs Exonum node with Java runtime enabled.

USAGE:
    exonum-java <SUBCOMMAND>

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information

SUBCOMMANDS:
    finalize             Generate final node configuration using public configs of other nodes in the network.
    generate-config      Generate public and private configs of the node.
    generate-template    Generate common part of the nodes configuration.
    help                 Prints this message or the help of the given subcommand(s)
    maintenance          Perform different maintenance actions.
    run                  Run the node with provided node config and Java runtime enabled.
```

Information is automatically parsed from `Cargo.toml` and docstrings.